File: Semester 4/Medii de proiectare si programare/Laborator/Laborator 4/FestivalMuzicaJava/src/main/java/domain/Show.java
package domain;

import lombok.Getter;
import lombok.Setter;

import java.time.LocalDate;
import java.util.Objects;

@Getter
@Setter
public class Show implements Entity<Long> {
    private Long id;
    private String artistName;
    private LocalDate date;
    private String location;
    private int remainingSeats;

    public Show(Long id, String artistName, LocalDate date, String location, int remainingSeats) {
        this.id = id;
        this.artistName = artistName;
        this.date = date;
        this.location = location;
        this.remainingSeats = remainingSeats;
    }

    @Override
    public Long getId() {
        return id;
    }

    @Override
    public void setId(Long id) {
        this.id = id;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        Show show = (Show) o;
        return remainingSeats == show.remainingSeats && Objects.equals(id, show.id) && Objects.equals(artistName, show.artistName) && Objects.equals(date, show.date) && Objects.equals(location, show.location);
    }

    @Override
    public int hashCode() {
        return Objects.hash(id, artistName, date, location, remainingSeats);
    }

    @Override
    public String toString() {
        return "Show{" +
                "id=" + id +
                ", artistName='" + artistName + '\'' +
                ", date=" + date +
                ", location='" + location + '\'' +
                ", remainingSeats=" + remainingSeats +
                '}';
    }
}

File: Semester 4/Medii de proiectare si programare/Laborator/Laborator 8/FestivalMuzicaCSharpWFA/model/Show.cs
namespace FestivalMuzicaCSharp.Domain;

public class Show : IEntity<long>
{
    public long Id { get; set; }
    public string ArtistName { get; set; }
    public DateOnly Date { get; set; }
    public string Location { get; set; }
    public int RemainingSeats { get; set; }
    
    public Show(long id, string artistName, DateOnly date, string location, int remainingSeats)
    {
        Id = id;
        ArtistName = artistName;
        Date = date;
        Location = location;
        RemainingSeats = remainingSeats;
    }

    public override bool Equals(object? obj) => Equals(obj as Show);

    public bool Equals(Show? other)
    {
        if (other is null) return false;
        if (ReferenceEquals(this, other)) return true;

        return Id == other.Id && string.Equals(ArtistName, other.ArtistName, StringComparison.Ordinal)
                              && string.Equals(Location, other.Location, StringComparison.Ordinal)
                              && Date.Equals(other.Date)
                              && RemainingSeats == other.RemainingSeats;
    }
    
    public override int GetHashCode() => HashCode.Combine(Id, ArtistName, Date, Location, RemainingSeats);

    public override string ToString()
    {
        return $"Show{{id={Id}, artistName='{ArtistName}', date={Date}, location='{Location}', remainingSeats={RemainingSeats}}}";
    }
}